commands/channel_e2e_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package commands
+
+import (
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/spf13/cobra"
+
+	"github.com/mattermost/mmctl/client"
+	"github.com/mattermost/mmctl/printer"
+)
+
+func (s *MmctlE2ETestSuite) TestListChannelsCmdF() {
+	s.SetupTestHelper().InitBasic()
+
+	s.Run("List channels/Client", func() {
+		printer.Clean()
+
+		err := listChannelsCmdF(s.th.Client, &cobra.Command{}, []string{s.th.BasicTeam.Name})
+		s.Require().Nil(err)
+		s.Equal(6, len(printer.GetLines()))
+		s.assertOutputDisplaysChannels()
+		s.Len(printer.GetErrorLines(), 0)
+	})
+
+	s.RunForSystemAdminAndLocal("List channels", func(c client.Client) {
+		printer.Clean()
+
+		err := listChannelsCmdF(c, &cobra.Command{}, []string{s.th.BasicTeam.Name})
+		s.Require().Nil(err)
+		s.Equal(7, len(printer.GetLines()))
+		s.assertOutputDisplaysChannels()
+		s.Len(printer.GetErrorLines(), 0)
+	})
+
+	s.RunForAllClients("List channels for non existent team", func(c client.Client) {
+		printer.Clean()
+		team := "non-existent-team"
+
+		err := listChannelsCmdF(c, &cobra.Command{}, []string{team})
+		s.Require().Nil(err)
+		s.Len(printer.GetErrorLines(), 1)
+		s.Equal("Unable to find team '"+team+"'", printer.GetErrorLines()[0])
+	})
+}
+
+func (s *MmctlE2ETestSuite) assertOutputDisplaysChannels() {
```

Comment:
I like the approach of building a list, but I'd avoid to make it a function on the test suite. Would you be open to create the list at the beginning of the `TestListChannelsCmdF` function and run one loop per test case??

For completion, in the test case of the unprivileged client with 6 channels in the response, I'd assert as well that the 7nth channel is not part of the list.

Reply:
Thanks a lot for this change, I think a function inside the scope of the tests makes a lot of sense.